components/metrics/http.go

```diff
@@ -0,0 +1,46 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/go-chi/chi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// ServeHTTP establishes an HTTP server that exposes the /metrics endpoint for Prometheus at the given address.
+// It returns the prometheus registry (to register the metrics on) and a canceling function that ends the server.
+func ServeHTTP(addr string) (registry *prometheus.Registry, cancel func()) {
```

Comment:
I think it voids SRP a bit

Reply:
671f1a6d0e6e14b3c9be0feb407e733c84d0bec3